addons/enemybar/enemybar.lua

```diff
@@ -24,31 +24,524 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+table = require('table')
+require 'bars'
+require 'actionTracking'
 
-player_id = 0
+player_id = nil
 debug_string = ''
 
-windower.register_event('load', function()
+local state = {}
+state.setup = false
+state.focustarget = nil
+
+function initialize_bars()
+    if target_bar then 
+        bars.destroy(target_bar) 
+        target_bar = nil
+    end
+    if subtarget_bar then 
+        bars.destroy(subtarget_bar)
+        subtarget_bar = nil 
+    end
+    if focustarget_bar then 
+        bars.destroy(focustarget_bar) 
+        focustarget_bar = nil
+    end
+    if aggro_bars then
+        for i,b in ipairs(aggro_bars) do
+            bars.destroy(b)
+        end
+    end
+
+    target_bar = bars.new(nil, settings.target_bar)
+    subtarget_bar = bars.new(nil, settings.subtarget_bar)
+    focustarget_bar = bars.new(nil, settings.focustarget_bar)
+    local y = settings.aggro_bar.pos.y
+    aggro_bars = {}
+    for i = 1, settings.aggro_bar.count do
+        aggro_bars[i] = bars.new(nil, settings.aggro_bar)
+        bars.move(aggro_bars[i], settings.aggro_bar.pos.x, y)
+        if settings.aggro_bar.stack_dir == 'down' then
+            y = y + settings.aggro_bar.stack_padding
+        elseif settings.aggro_bar.stack_dir == 'up' then
+            y = y - settings.aggro_bar.stack_padding
+        end
+    end
+
+    bar_sets = {{target_bar}, {subtarget_bar}, {focustarget_bar}, aggro_bars}
+end
+
+function update_bar(bar, target, show)
+    if state.setup then
+        if show then
+            bars.show(bar)
+            if bar == target_bar then bars.update_target(bar, "Target Name", 79, 12.1, 1)
+            elseif bar == subtarget_bar then bars.update_target(bar, "Subtarget Name", 53, 11.4, 2)
+            elseif bar == focustarget_bar then bars.update_target(bar, "Focus Target Name", 36, 8.6, 1)
+            else bars.update_target(bar, "Aggro Target Name", 47, 6.6, 1) end
+            bars.update_action(bar, "Action Name", '')
+            bars.update_enmity(bar, "Mob Target", {red=102, green=255, blue=255})
+            bars.update_status(bar, {})
+            bars.set_name_color(bar, {red=255, green=180, blue=180})
+        else
+            bars.hide(bar)
+        end
+    else
+        if target ~= nil and show then     
+            bars.show(bar)
+
+            local dist = get_distance(windower.ffxi.get_mob_by_target('me'), target)
+
+            local t = windower.ffxi.get_mob_by_target('t')
+            local st = windower.ffxi.get_mob_by_target('st')
+            local target_type = nil
+            if t and t.id == target.id then 
+                target_type = 1
+            elseif st and st.id == target.id then
+                target_type = 2
+            end
+            bars.update_target(bar, target.name, target.hpp, dist, target_type)
+
+            local action = tracked_actions[target.id]
+            if action and not action.complete then
+                    bars.update_action(bar, action.ability.en, '')
```

Comment:
Use `.name` instead of `.en` (and use `.english` over `.en` in any case). `.name` will resolve to `.english` on NA clients and to `.japanese` on JP clients.